### ISSUE TITLE
Update setup.py script

### ISSUE THREAD

**Issue by erikvansebille** [OceanParcels/parcels#104]
The `setup.py` script in the repo root is fairly outdated. Now that parcels is becoming bigger and installation is becoming more subtle, we should update this script. 

Things that `setup.py` should do
- check for proper versions of netcdf etc
- try install `matplotlib` and `Basemap` (and gracefully fail)
- `pip install requirements`
- run the `pull_data` script (with the option not to do this if on low-bandwidth, perhaps by asking users whether they want to download the auxiliary data)
- set the `PYTHONPATH` to include parcels

**Comment by mlange05** [comment]
Ok, I think this needs a little clarification: `setup.py` is a standardised install script that is required by `pip` to install packages in the local Python environment. As such there is no need for it to set `PYTHONPATH` since it will install Parcels in a central location that Python always searches during any `import`.

As for the dependencies, the standardised `setuptools` utilities allow dependencies to be specified for `pip install` but convention states that the these should be **minimal**, ie. they should not include `matplotlib`, `Basemap` or `pull_data`. `requitements.txt` on the other hand is meant to fully define a kind of "best known" environment, so we probably should add `matplotlib` and `Basemap` here. More info can be found [here](https://packaging.python.org/requirements/).

So, I don't think the "do-it-all-out-of-the-box" install script you are describing here is what `setup.py` is meant to do. One way to do this would be to add a dedicated project-specific script like `parcels-install` that does all the additional sanity checking, like netCDF versions and pulling example data. To me this sounds a bit more like renaming and adding a few features to `pull_data`, and I'll look into that.

**Comment by fischcheng** [comment]
It sounds to me like a install and test script to ensure all data are ready for the tutorials, so just like the last few lines in `.travis.yml`

**Comment by rabernat** [comment]
This seems like the relevant place to repeat my comment from #225.

[Conda forge](https://conda-forge.org/) will likely solve your binary dependency and distributions problems. That's why conda forge was created.

**Comment by willirath** [comment]
> Conda forge will likely solve your binary dependency and distributions problems. That's why conda forge was created.

I agree. From the discussion above and from what I tried in #225, I got the impression that at this stage, a parcels package in conda-forge might not make a lot of sense, though. Users might be more likely to work with the bleeding edge from master or even some commit from a PR.

For this (and from my PoV...), a combination of a `setup.py` and a conda-env heavily relying on conda-forge would be a good compromise.
